### ISSUE TITLE
False positive with dataclass multiple inheritance and field init

### ISSUE THREAD

**Issue by tomsilver** [PyCQA/pylint#7427]
### Bug description

`pylint` emits a false positive `too-many-function-args` in the following situation:

```python
from dataclasses import dataclass, field

@dataclass
class A:
    _foo: bool = field(init=False)

    def __post_init__(self) -> None:
        self._foo = True


@dataclass
class B:
    bar: bool


@dataclass
class C(A, B):
    pass


a = A()
assert a._foo  # no error

b = B(False)
assert not b.bar  # no error

c = C(False)  # too-many-function-args
assert not c.bar
assert c._foo

```

**Note:** This is new in `pylint==2.15.0`. There is no false positive with `pylint==2.14.5`.

### Command used

```shell
pylint --disable all --enable too-many-function-args example.py
```

### Pylint output

```shell
************* Module bug
bug.py:27:4: E1121: Too many positional arguments for constructor call (too-many-function-args)

------------------------------------------------------------------
Your code has been rated at 6.88/10 (previous run: 6.88/10, +0.00)
```

### Expected behavior

`pylint` should not raise `too-many-function-args`

### Pylint version

```shell
pylint 2.15.0
astroid 2.12.8
Python 3.9.4 (default, Aug  1 2022, 10:10:32)
```

### OS / Environment

macOS Monterey 12.5.1

**Comment by DanielNoord** [comment]
Thanks for the report! A fix should be available somewhere today/tonight when we release a new version of `pylint` and `astroid` 😄
